miio/toiletlid.py

```diff
@@ -0,0 +1,147 @@
+import enum
+import logging
+from typing import Any, Dict
+
+import click
+
+from .click_common import command, format_output, EnumType
+from .device import Device
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_TOILETLID_V1 = "tinymu.toiletlid.v1"
+
+AVAILABLE_PROPERTIES_COMMON = ["work_state", "filter_use_flux", "filter_use_time"]
+
+AVAILABLE_PROPERTIES = {MODEL_TOILETLID_V1: AVAILABLE_PROPERTIES_COMMON}
+
+
+class AmbientLightColor(enum.Enum):
+    White = "0"
+    Yellow = "1"
+    Powder = "2"
+    Green = "3"
+    Purple = "4"
+    Blue = "5"
+    Orange = "6"
+    Red = "7"
+
+
+class ToiletlidStatus:
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def work_state(self) -> int:
+        """Device state code"""
+        return self.data["work_state"]
+
+    @property
+    def is_on(self) -> bool:
+        return self.work
+
+    @property
+    def work(self) -> bool:
+        """True if device is use on."""
+        return self.work_state != 1
+
+    @property
+    def filter_use_percentage(self) -> str:
+        """Filter percentage of remaining life"""
+        return "{}%".format(self.data["filter_use_flux"])
+
+    @property
+    def filter_remaining_time(self) -> int:
+        """Filter remaining life days"""
+        return self.data["filter_use_time"]
+
+    @property
+    def ambient_light(self) -> str:
+        """Ambient light color."""
+        return self.data["ambient_light"]
+
+    def __repr__(self) -> str:
+        return (
+            "<ToiletlidStatus work=%s, "
+            "state=%s, "
+            "ambient_light=%s, "
+            "filter_use_percentage=%s, "
+            "filter_remaining_time=%s>"
+            % (
+                self.work,
+                self.work_state,
+                self.ambient_light,
+                self.filter_use_percentage,
+                self.filter_remaining_time,
+            )
+        )
+
+
+class Toiletlid(Device):
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MODEL_TOILETLID_V1,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in AVAILABLE_PROPERTIES:
+            self.model = model
+        else:
+            self.model = MODEL_TOILETLID_V1
```

Comment:
Do you think there'll be more models than this one? Because if not, the code could be made much simpler by aoiding the model handling altogether.

Reply:
There are two brands and three models with similar functions. They are `Tiny mu smart toiletdled` `Tiny mu smart toiletdled voice AI version` `little whale wash smart toilet pro`

Reply:
Ok, thanks for being farsighted! :+1: